apps/studio/src/components/ResourceSelector/ResourceItem.tsx

```diff
@@ -0,0 +1,109 @@
+import type { IconType } from "react-icons"
+import { Suspense, useMemo } from "react"
+import { Box, HStack, Icon, Mark, Skeleton, Text } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { QueryErrorResetBoundary } from "@tanstack/react-query"
+import fuzzysort from "fuzzysort"
+import { ErrorBoundary } from "react-error-boundary"
+import { BiData, BiFile, BiFolder, BiHome, BiLockAlt } from "react-icons/bi"
+
+import type { RouterOutput } from "~/utils/trpc"
+
+type ResourceItemProps = Pick<
+  RouterOutput["resource"]["getChildrenOf"][number],
+  "permalink" | "type"
+> & {
+  isSelected: boolean
+  isDisabled: boolean
+  searchQuery: string
+  onResourceItemSelect: () => void
+}
+
+const SuspendableResourceItem = ({
+  permalink,
+  type,
+  isSelected,
+  isDisabled,
+  searchQuery,
+  onResourceItemSelect,
+}: ResourceItemProps) => {
+  const icon: IconType = useMemo(() => {
+    switch (type) {
+      case "Folder":
+        return BiFolder
+      case "CollectionPage":
+      case "Page":
+        return BiFile
+      case "Collection":
+        return BiData
+    }
+  }, [type])
+
+  const SearchTextHighlight = useMemo(() => {
+    const result = fuzzysort.single(searchQuery, permalink)
+    if (!result) {
+      return permalink
+    }
+
+    return fuzzysort.highlight(result, (m, i) => (
+      <Mark key={i} bgColor="interaction.tinted.main.active">
+        {m}
+      </Mark>
+    ))
+  }, [searchQuery, permalink])
+
+  return (
+    <Button
+      variant="clear"
```

Comment:
`variant="unstyled"` may be a closer match and don't require you to override so many things, or it'll become a monster

Reply:
Hmm `unstyled` might require even more stylings I think, the button itself currently is still quite close to the clear variant so opted to keep it.